ctapipe/tools/stage1.py

```diff
@@ -533,35 +483,101 @@ def _parameterize_image(self, subarray, data, tel_id):
+            timing = timing_parameters(
                 geom=geom_selected,
                 image=image_selected,
-                peak_time=data.peak_time[signal_pixels],
-                hillas_parameters=params.hillas,
+                peak_time=dl1_camera.peak_time[signal_pixels],
+                hillas_parameters=hillas,
             )
-            params.leakage = leakage(
-                geom=geometry, image=data.image, cleaning_mask=signal_pixels
+            leakage = leakage_parameters(
+                geom=geometry, image=dl1_camera.image, cleaning_mask=signal_pixels
             )
-            params.concentration = concentration(
+            concentration = concentration_parameters(
                 geom=geom_selected,
                 image=image_selected,
-                hillas_parameters=params.hillas,
+                hillas_parameters=hillas,
             )
-            params.morphology = morphology(
+            morphology = morphology_parameters(
                 geom=geometry, image_mask=signal_pixels
             )
-            params.intensity_statistics = descriptive_statistics(
+            intensity_statistics = descriptive_statistics(
                 image_selected, container_class=IntensityStatisticsContainer
             )
-            params.peak_time_statistics = descriptive_statistics(
-                data.peak_time[signal_pixels],
+            peak_time_statistics = descriptive_statistics(
+                dl1_camera.peak_time[signal_pixels],
                 container_class=PeakTimeStatisticsContainer,
             )
 
-        return signal_pixels, params
+            return signal_pixels, ImageParametersContainer(
+                hillas=hillas,
+                timing=timing,
+                leakage=leakage,
+                morphology=morphology,
+                concentration=concentration,
+                intensity_statistics=intensity_statistics,
+                peak_time_statistics=peak_time_statistics,
+            )
+
+        # return the default container (containing nan values) for no
+        # parameterization
+        return signal_pixels, ImageParametersContainer()
+
+    def _parameterize_true_image(self, tel_id, true_image):
+        """Apply image cleaning and calculate image features
+
+        Parameters
+        ----------
+        subarray : SubarrayDescription
+           subarray description
+        data : DL1CameraContainer
+            calibrated camera data
+        tel_id: int
+            which telescope is being cleaned
+
+        Returns
+        -------
+        np.ndarray, ImageParametersContainer:
+            cleaning mask, parameters
+        """
+
+        tel = self.event_source.subarray.tel[tel_id]
+        geometry = tel.camera.geometry
+
+        # apply cleaning
+        signal_pixels = true_image > 0
+        image_selected = true_image[signal_pixels]
+
+        # parameterize the event if all criteria pass:
```

Comment:
the rest of this could probably be turned into a sub-function, since it's duplicated code from `parameterize_image`, with only image_selected being different.

Reply:
Not really, since the true image does not have timing information. But I can split it up in things that need timing information and in things that don't.

Reply:
Do you think it might be worth it to ask Konrad about providing a mean photon arrival time for the true photons in each pixel?

Reply:
No, I hadn't thought of that, it's ok for now - we can think of making more functions later.  

It would be nice to have the arrival time in the MCs though! Would make testing easier.